future-apps/payroll/contracts/Payroll.sol

```diff
@@ -738,4 +755,34 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         Employee storage employee = employees[_employeeId];
         return employee.endDate >= getTimestamp64();
     }
+
+    /**
+     * @dev Updates the accrued salary and payroll date of a given employee based on a paying amount and their current owed salary
+     * @param _employeeId Employee's identifier
+     * @param _payingAmount Amount being payed to the employee
+     * @param _currentOwedSalary Amount owed to the employee for his current salary
+     */
+    function _updateEmployeeStatusBasedOnPaidPayroll(uint256 _employeeId, uint256 _payingAmount, uint256 _currentOwedSalary) private {
+        Employee storage employee = employees[_employeeId];
+        uint256 accruedSalary = employee.accruedSalary;
+
+        if (_payingAmount <= accruedSalary) {
+            // employee is only cashing out some previous owed salary, then we don't need to update the last payroll date
+            // there's no need to use safemath here since we already know that accruedSalary is >= than _payingAmount
+            employee.accruedSalary = accruedSalary - _payingAmount;
+        }
+        else if (accruedSalary == 0) {
```

Comment:
Small nitpick: we generally put the `else if` on the same line a the closing brace